src/main/php/PDepend/Source/Language/PHP/PHPParserVersion56.php

```diff
@@ -333,24 +333,14 @@ protected function parseArgumentList(ASTArguments $arguments)
     {
         while (true) {
             $this->consumeComments();
+
             if (Tokens::T_ELLIPSIS === $this->tokenizer->peek()) {
                 $this->consumeToken(Tokens::T_ELLIPSIS);
             }
 
-            $this->consumeComments();
-            if (null === ($expr = $this->parseOptionalExpression())) {
+            if (!(($expr = $this->parseOptionalExpression()) && $this->addChildToList($arguments, $expr))) {
```

Comment:
Maybe I just don't like assignments inside statements, but to me it just leads to excessive parenthecies.
```suggestion
            $expr = $this->parseOptionalExpression();
            if (!$expr || !$this->addChildToList($arguments, $expr)) {
```